compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -312,8 +312,11 @@ object SpaceEngine {
    */
   def isIrrefutable(unapp: tpd.Tree, argLen: Int)(using Context): Boolean = {
     val fun1 = tpd.funPart(unapp)
-    val funRef = fun1.tpe.asInstanceOf[TermRef]
```

Comment:
is there something that specifically failed to prompt this change?

Reply:
Yes, some tests started crashing.  This change is in a separate commit (c1b83abd5b276a95ad67a81683a4d692c12747d6) and the motivating details are in the commit message.  Basically, `fun1.tpe` can be an instance of `ErrorType` when the function is not a valid extractor.

But I agree that the code change itself is not clear, I should probably at least write `case _: ErrorType =>`